#VRML_SIM R2021b utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation -0.5395119695430545 0.6633482102231956 0.5185518168066303 1.9216154985983016
  position 0.6624067114240257 3.726091089300381 0.43266367646476894
}
TexturedBackground {
}
TexturedBackgroundLight {
}
DEF area RectangleArena {
  translation -0.260407 -4.44089e-16 -0.190988
  floorSize 2.286 2.794
  floorTileSize 1.016 1.016
}
DEF YellowCylinder Solid {
  translation -0.281701 0.2 -1.14024
  rotation 1.92079e-07 -1 1.10737e-06 1.30899
  children [
    DEF landmark Shape {
      appearance PBRAppearance {
        baseColor 1 1 0
        roughness 1
        metalness 0
      }
      geometry Cylinder {
        height 0.7
        radius 0.15
      }
    }
  ]
  boundingObject USE landmark
  physics DEF landmarkPH Physics {
  }
  recognitionColors [
    0 0 1
  ]
}
WoodenBox {
  translation -0.221274 0.0300054 0.31901
  rotation -1.9085499999932805e-14 0.9999999999964793 -2.6535799999906574e-06 3.14159
  name "wooden box(2)"
  size 0.642 0.05 0.254
}
WoodenBox {
  translation -0.215559 0.0299992 -0.082969
  rotation -1.9085499999932805e-14 0.9999999999964793 -2.6535799999906574e-06 3.14159
  name "wooden box(4)"
  size 0.128 0.05 0.558
}
WoodenBox {
  translation 0.224441 0.0300037 0.447031
  rotation -1.9085499999932805e-14 0.9999999999964793 -2.6535799999906574e-06 3.14159
  name "wooden box(3)"
  size 0.254 0.05 0.508
}
WoodenBox {
  translation -0.655559 0.0300036 0.44703
  rotation -1.9085499999932805e-14 0.9999999999964793 -2.6535799999906574e-06 3.14159
  name "wooden box(1)"
  size 0.254 0.05 0.508
}
WoodenBox {
  translation -0.325559 0.0299969 -0.48297
  rotation 2.653579999981315e-06 -0.9999999999929585 2.653579999981315e-06 -1.5707953071795862
  name "wooden box(5)"
  size 0.254 0.05 1.137
}
DEF epuck E-puck {
  translation 0.306738 4.79175e-05 1.13453
  rotation -0.000399516968110873 -0.9999999201808009 -4.9582396042372526e-06 1.5708053071795867
  controller "lab2_task4"
  supervisor TRUE
  camera_width 50
  turretSlot [
    InertialUnit {
      rotation 0 1 0 3.14
    }
    DEF righthinge HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "right wheel motor"
          consumptionFactor 70
          controlPID 8 2 0.5
        }
        PositionSensor {
          name "right wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0.04500995778380097 0.025013380758883635 -2.4671929852070235e-05
        rotation 0.9999993203499804 0.0010052734270078168 -0.0005905293508524928 4.6368371220546685
        children [
          DEF WHEEL Transform {
            translation -0.007 0.01 -0.05
            rotation 0 0 1 1.57
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 0 0 0
                  transparency 1
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.02
                }
              }
            ]
            rotationStep 0.25
          }
        ]
        name "right wheel(1)"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
        rotationStep 0.25
      }
    }
    DEF lefthinge HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "left wheel motor"
          consumptionFactor 70
          controlPID 8 0 0
        }
        PositionSensor {
          name "left wheel sensor"
        }
      ]
      endPoint Solid {
        translation -0.04501012942703836 0.024977671977588888 -4.5733976313299836e-05
        rotation -0.9999982113394998 0.0004727526945777002 -0.0018313445035503763 1.2425810135142943
        children [
          DEF WHEEL Transform {
            translation 0.007 0 -0.05
            rotation 0 0 1 1.57
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 0.1 0.1 0.1
                  transparency 1
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.02
                }
              }
            ]
            rotationStep 0.25
          }
        ]
        name "left wheel(1)"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
      }
    }
    DEF front_ds DistanceSensor {
      translation 0 0 -0.045
      rotation 0 1 0 1.5708
      name "front_ds"
      recognitionColors [
        0 0 0
      ]
      lookupTable [
        0 0 0
        1.27 1.27 0
      ]
    }
    DEF right_ds DistanceSensor {
      translation 0.035 -0.01 -0.01
      name "right_ds"
      lookupTable [
        0 0 0
        1.27 1.27 0
      ]
    }
    DEF left_ds DistanceSensor {
      translation -0.035 -0.01 -0.01
      rotation 0 1 0 3.14159
      name "left_ds"
      lookupTable [
        0 0 0
        1.27 1.27 0
      ]
    }
    Camera {
      translation 0 -0.01 -0.045
      name "camera1"
      recognitionColors [
        1 1 0
      ]
      fieldOfView 1.57
      width 80
      height 80
      recognition Recognition {
      }
    }
  ]
}
